### PR TITLE
Field map obj

### PR DESCRIPTION
**Pull request by scnorton**

### CODE REVIEW

**Comment by yenatch** [comment]
```
arm-none-eabi-ld.exe: Warning: alignment 4 of symbol `gUnknown_084975C4' in data/field_map_obj.o is smaller than 16 in src/field_map_obj.o
```

**Comment by yenatch** [comment]
```
arm-none-eabi-ld.exe: Warning: alignment 4 of symbol `gUnknown_084975C4' in data/field_map_obj.o is smaller than 16 in src/field_map_obj.o
arm-none-eabi-ld.exe: Warning: alignment 4 of symbol `gUnknown_084975D4' in data/field_map_obj.o is smaller than 16 in src/field_map_obj.o
arm-none-eabi-ld.exe: Warning: alignment 4 of symbol `gUnknown_084975EC' in data/field_map_obj.o is smaller than 16 in src/field_map_obj.o
arm-none-eabi-ld.exe: Warning: alignment 4 of symbol `gUnknown_08505438' in data/field_map_obj.o is smaller than 16 in src/field_map_obj.o
arm-none-eabi-ld.exe: Warning: alignment 4 of symbol `gUnknown_0850557C' in data/field_map_obj.o is smaller than 16 in src/field_map_obj.o
arm-none-eabi-ld.exe: Warning: alignment 1 of symbol `gUnknown_085055CD' in data/field_map_obj.o is smaller than 16 in src/field_map_obj.o
arm-none-eabi-ld.exe: Warning: alignment 4 of symbol `gFieldObjectGraphicsInfoPointers' in data/field_map_obj.o is smaller than 16 in src/field_map_obj.o
arm-none-eabi-ld.exe: Warning: alignment 4 of symbol `gMauvilleOldManGraphicsInfoPointers' in data/field_map_obj.o is smaller than 16 in src/field_map_obj.o
arm-none-eabi-ld.exe: Warning: alignment 4 of symbol `gUnknown_0850BBC8' in data/field_map_obj.o is smaller than 16 in src/field_map_obj.o
arm-none-eabi-ld.exe: Warning: alignment 4 of symbol `gUnknown_0850BD00' in data/field_map_obj.o is smaller than 16 in src/field_map_obj.o
arm-none-eabi-ld.exe: Warning: alignment 4 of symbol `gUnknown_0850BD78' in data/field_map_obj.o is smaller than 16 in src/field_map_obj.o
arm-none-eabi-ld.exe: Warning: alignment 4 of symbol `gUnknown_0850BE38' in data/field_map_obj.o is smaller than 8 in src/field_map_obj.o
arm-none-eabi-ld.exe: Warning: alignment 4 of symbol `gUnknown_0850D6DC' in data/field_map_obj.o is smaller than 8 in src/field_map_obj.o
arm-none-eabi-ld.exe: Warning: alignment 4 of symbol `gUnknown_0850D6EC' in data/field_map_obj.o is smaller than 8 in src/field_map_obj.o
arm-none-eabi-ld.exe: Warning: alignment 4 of symbol `gUnknown_0850D714' in data/field_map_obj.o is smaller than 16 in src/field_map_obj.o
arm-none-eabi-ld.exe: Warning: alignment 4 of symbol `gUnknown_0850D8AC' in data/field_map_obj.o is smaller than 8 in src/field_map_obj.o
arm-none-eabi-ld.exe: Warning: alignment 4 of symbol `gUnknown_0850D8C4' in data/field_map_obj.o is smaller than 8 in src/field_map_obj.o
arm-none-eabi-ld.exe: Warning: alignment 4 of symbol `gUnknown_0850DA64' in data/field_map_obj.o is smaller than 16 in src/field_map_obj.o
arm-none-eabi-ld.exe: Warning: alignment 4 of symbol `gUnknown_0850DB5C' in data/field_map_obj.o is smaller than 16 in src/field_map_obj.o
arm-none-eabi-ld.exe: Warning: alignment 4 of symbol `gUnknown_0850DB6C' in data/field_map_obj.o is smaller than 16 in src/field_map_obj.o
arm-none-eabi-ld.exe: Warning: alignment 4 of symbol `gUnknown_0850DB7C' in data/field_map_obj.o is smaller than 16 in src/field_map_obj.o
arm-none-eabi-ld.exe: Warning: alignment 4 of symbol `gUnknown_0850DBA0' in data/field_map_obj.o is smaller than 8 in src/field_map_obj.o
arm-none-eabi-ld.exe: Warning: alignment 1 of symbol `gUnknown_0850DC2F' in data/field_map_obj.o is smaller than 16 in src/field_map_obj.o
arm-none-eabi-ld.exe: Warning: alignment 1 of symbol `gUnknown_0850DC3F' in data/field_map_obj.o is smaller than 16 in src/field_map_obj.o
arm-none-eabi-ld.exe: Warning: alignment 4 of symbol `gUnknown_0850DC50' in data/field_map_obj.o is smaller than 16 in src/field_map_obj.o
arm-none-eabi-ld.exe: Warning: alignment 4 of symbol `gUnknown_0850DEE8' in data/field_map_obj.o is smaller than 16 in src/field_map_obj.o
arm-none-eabi-ld.exe: Warning: alignment 4 of symbol `gUnknown_0850DFBC' in data/field_map_obj.o is smaller than 8 in src/field_map_obj.o
arm-none-eabi-ld.exe: Warning: alignment 2 of symbol `gUnknown_0850DFC2' in data/field_map_obj.o is smaller than 8 in src/field_map_obj.o
```

File: data/field_map_obj.s [hunk 1]
```diff
@@ -322,481 +322,1408 @@ Unknown_0850BD70:: @ 0x0850bd70
...
+	.byte 0x02, 0x01, 0x04, 0x03, 0x01, 0x02, 0x03, 0x04, 0x04, 0x03, 0x01, 0x02, 0x03, 0x04, 0x02, 0x01, 0x00
 
+	.align 2, 0
 gUnknown_0850DC50:: @ 850DC50
-	.incbin "baserom.gba", 0x50dc50, 0x298
+@ replacing .incbin "baserom.gba", 0x0050dc50, 0x298
+	.4byte gUnknown_0850DC50 + 0x278
+	.4byte gUnknown_0850DC50 + 0x280
+	.4byte gUnknown_0850DC50 + 0x288
+	.4byte gUnknown_0850DC50 + 0x290
```
Comment (yenatch) [inline data/field_map_obj.s]: this seems wrong

File: asm/battle_dome_cards.s [hunk 2]
```diff
@@ -1277,5 +1277,4 @@ _0818D9A2:
 	bx r1
 	.pool
 	thumb_func_end sub_818D97C
-
```
Comment (ProjectRevoTPP) [inline asm/battle_dome_cards.s]: was this really necessary? why remove the empty line?

File: include/global.fieldmap.h [hunk 3]
```diff
@@ -224,7 +226,13 @@ struct MapObject
...
     /*0x18*/ u8 mapobj_unk_18:4;  //current direction?
     /*0x18*/ u8 placeholder18:4;
-    /*0x19*/ u8 mapobj_unk_19;
+    /*0x19*/ union __attribute__((packed)) {
+        u8 as_byte;
+        struct __attribute__((packed)) {
+            u8 x:4;
+            u8 y:4;
+        } __attribute__((aligned (1))) as_nybbles;
+    } __attribute__((aligned (1))) range;
```
Comment (yenatch) [inline include/global.fieldmap.h]: I like this naming scheme.

File: asm/battle_dome_cards.s (hunk 2 above)
Comment (yenatch) [reply asm/battle_dome_cards.s]: Probably automatic. Not sure why this is seen as a bad thing.

File: data/field_map_obj.s (hunk 1 above)
Comment (yenatch) [reply data/field_map_obj.s]: oh, it was fixed

**Pull request merged**